core/templates/pages/about-page/about-page.component.spec.ts

```diff
@@ -55,165 +59,186 @@ class MockI18nLanguageCodeService {
+
+  it('should successfully instantiate the component',
+    () => {
+      expect(component).toBeDefined();
+      component.ngOnInit();
+    });
+
+  it('should return correct static image url when calling getStaticImageUrl',
+    () => {
+      expect(component.getStaticImageUrl('/path/to/image')).toBe(
+        '/assets/images/path/to/image');
+    });
+
+  // eslint-disable-next-line max-len
+  it('should redirect guest user to the login page when they click create lesson button',
+    fakeAsync(() => {
+      const userInfoBackendDict = {
+        is_moderator: false,
+        is_admin: false,
+        is_super_admin: false,
+        is_topic_manager: false,
+        can_create_collections: false,
+        preferred_site_language_code: null,
+        username: '',
+        email: '',
+        user_is_logged_in: null
+      };
+      spyOn(userService, 'getUserInfoAsync').and.returnValue(Promise.resolve(
+        UserInfo.createFromBackendDict(userInfoBackendDict))
+      );
+      component.ngOnInit();
+      flushMicrotasks();
+      expect(component.userIsLoggedIn).toBe(null);
+      spyOn(
+        siteAnalyticsService, 'registerCreateLessonButtonEvent')
+        .and.callThrough();
+      component.onClickCreateLessonButton();
+
+      expect(siteAnalyticsService.registerCreateLessonButtonEvent)
+        .toHaveBeenCalledWith();
+      expect(component.loginUrl).toBe('/_ah/login');
+    }));
+
+  it('should set component properties when ngOnInit() is called', () => {
     component.ngOnInit();
 
-    expect(windowRef.nativeWindow.location.hash).toBe('#about');
-    expect(component.activeTabName).toBe('about');
+    expect(component.userIsLoggedIn).toBe(false);
+    expect(component.classroomUrl).toBe('/learn/math');
+    expect(component.loginUrl).toBe('/_ah/login');
   });
 
-  it('should activate about license on init', () => {
-    windowRef.nativeWindow.location.hash = '#license';
+  it('should show and hide loading screen with the correct text', () =>
+    fakeAsync(() => {
+      component.ngOnInit();
+      spyOn(loaderService, 'showLoadingScreen').and.callThrough();
+      expect(loaderService.showLoadingScreen)
+        .toHaveBeenCalledWith('Loading');
+    }));
+
+  it('should set the correct value for the userIsLoggedIn property',
+    fakeAsync(() => {
+      const UserInfoObject = {
+        is_moderator: false,
+        is_admin: false,
+        is_super_admin: false,
+        is_topic_manager: false,
+        can_create_collections: true,
+        preferred_site_language_code: null,
+        username: 'tester',
+        email: 'test@test.com',
+        user_is_logged_in: false
+      };
+      spyOn(userService, 'getUserInfoAsync').and.returnValue(Promise.resolve(
+        UserInfo.createFromBackendDict(UserInfoObject))
+      );
+
+      component.ngOnInit();
+      flushMicrotasks();
+
+      expect(component.userIsLoggedIn).toBe(false);
+    }));
+
+  it('should check that user is logged in', fakeAsync(() => {
```

Comment:
The test message for the above test and this test looks the same, can you please add a better test message?

Reply:
wrote another test message